### ISSUE TITLE
Can you update a rule tile appearance in a script? 

### ISSUE THREAD

**Issue by jatkinson16** [Unity-Technologies/2d-extras#311]
I've just started using rule tiles in Unity 2019.4 and I'm working on a procedurally generated tilemap that uses rule tiles. It use the standard `tilemaps.SetTile(new Vector3Int(x, y, 0), ruleTile);` to assign the tiles to tilemap positions but when the game runs the tiles don't adapt to those around them.

Is it possible to manually force a rule tile to recheck the tiles around it and update its sprite accordingly? I'm planning on having tiles get removed from tilemaps due to player actions in the future so that would also benefit from the answer.

**Comment by ChuanXin-Unity** [comment]
Hi, when you call SetTile with the RuleTile, the RuleTile should update all the positions around it, as defined in RuleTile.RefreshTile.

Could you share an example of where the RuleTile does not work and what Tiles are placed around the RuleTile?

> Is it possible to manually force a rule tile to recheck the tiles around it and update its sprite accordingly? I'm planning on having tiles get removed from tilemaps due to player actions in the future so that would also benefit from the answer.
You can call Tilemap.RefreshTile with the given positions to update those positions if you need to. However, the RuleTile system should update the Tiles automatically for you when you call SetTile.

**Comment by jatkinson16** [comment]
The rule tile works as normal in the tile palette
![image](https://user-images.githubusercontent.com/39330127/134906112-f74b54d3-6380-45e6-9f55-f39a6a7ad7b2.png)

But this is the outcome when the tiles are placed in a script(with the script looping from 0,0 to 56,32)
![image](https://user-images.githubusercontent.com/39330127/134905871-5857b7d2-9dfd-47a3-9b7e-95c280fc6c60.png)

I can assume that initially the tiles won't appear as they should when the script is still going through each coordinate placing tiles but their appearance ends up as the top corner tile regardless.
![image](https://user-images.githubusercontent.com/39330127/134906911-c52d817f-0dd2-47f2-ba51-ed907fd0c3fa.png)

That bottom left tile in the runtime image surely fulfils the condition for this rule tile, not the one it appears as
![image](https://user-images.githubusercontent.com/39330127/134907482-41d7f5a8-8c62-4987-a841-9023fd515033.png)

**Comment by jatkinson16** [comment]
I managed to figure it out thanks to checking with the sample rule tile. It was an issue on my end with how I set up different spritesheets, I removed the method call and the rule tiles were drawn properly. So it was just my fault in the end. Thanks for the help regardless, this package is a great tool for 2D games!
![image](https://user-images.githubusercontent.com/39330127/135048627-363f145b-908b-41bc-8f7c-a845f472fd26.png)
